lib/systems/doubles.nix

```diff
@@ -3,53 +3,53 @@ let
   inherit (lib) lists;
   inherit (lib.systems) parse;
   inherit (lib.systems.inspect) predicates;
-  inherit (lib.attrsets) matchAttrs;
+  inherit (lib.attrsets) attrValues matchAttrs;
 
-  all = [
-    "aarch64-linux"
-    "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux"
+  byOS = {
+    cygwin = [ "i686-cygwin" "x86_64-cygwin" ];
 
-    "mipsel-linux"
+    darwin = [ "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ];
 
-    "i686-cygwin" "i686-freebsd" "i686-linux" "i686-netbsd" "i686-openbsd"
+    freebsd = [ "i686-freebsd" "x86_64-freebsd" ];
 
-    "x86_64-cygwin" "x86_64-freebsd" "x86_64-linux"
-    "x86_64-netbsd" "x86_64-openbsd" "x86_64-solaris"
+    genode = [ "aarch64-genode" "i686-genode" "x86_64-genode" ];
 
-    "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin"
+    illumos = [ "x86_64-solaris" ];
 
-    "x86_64-windows" "i686-windows"
+    js = [ "js-ghcjs" ];
 
-    "wasm64-wasi" "wasm32-wasi"
+    linux = [
+      "aarch64-linux" "armv5tel-linux" "armv6l-linux" "armv7a-linux"
+      "armv7l-linux" "i686-linux" "mipsel-linux" "powerpc64-linux"
+      "powerpc64le-linux" "riscv32-linux" "riscv64-linux" "x86_64-linux"
+    ];
 
-    "x86_64-redox"
+    mmix = [ "mmix-mmixware" ];
```

Comment:
I think MMIX is the architecture's name, not the OS?

Reply:
You're absolutely right.  Fixed, and thanks for catching it.